api/client/client.go

```diff
@@ -275,26 +271,22 @@ func connect(ctx context.Context, cfg Config) (*Client, error) {
 // dialGRPC dials a connection between server and client. If withBlock is true,
 // the dial will block until the connection is up, rather than returning
 // immediately and connecting to the server in the background.
-func (c *Client) dialGRPC(ctx context.Context, addr string, withBlock bool) error {
-	dialOptions := []grpc.DialOption{
+func (c *Client) dialGRPC(ctx context.Context, addr string) error {
+	dialContext, cancel := context.WithTimeout(ctx, c.c.DialTimeout)
+	defer cancel()
+
+	dialOptions := append(
+		c.c.DialOpts,
 		grpc.WithContextDialer(c.grpcDialer()),
 		grpc.WithTransportCredentials(credentials.NewTLS(c.tlsConfig)),
-		grpc.WithKeepaliveParams(keepalive.ClientParameters{
-			Time:                c.c.KeepAlivePeriod,
-			Timeout:             c.c.KeepAlivePeriod * time.Duration(c.c.KeepAliveCount),
-			PermitWithoutStream: true,
-		}),
-	}
-
-	if withBlock {
-		dialOptions = append(dialOptions, grpc.WithBlock())
-	}
+	)
 
-	dialContext, cancel := context.WithTimeout(ctx, c.c.DialTimeout)
-	defer cancel()
+	// closedFlag should be set to 0 before dialing.
+	c.closedFlag = new(int32)
```

Comment:
set this where the `Client` struct is created - in `connect` and `connectInBackground`
a `newClient` constructor function might be handy for this